### ISSUE TITLE
Usage with path aliases

### ISSUE THREAD

**Issue by sami616** [wessberg/rollup-plugin-ts#96]
I've set up some custom path aliases in my ts-config: ie:

```json
"paths": {
    "@themes": ["themes"],
    "@themes/*": ["themes/*"],
  },
```

Running this through `rollup-plugin-ts` results in the below error

`[!] (plugin Typescript) TS2307: Cannot find module '@themes'.`

**Comment by antikvarBE** [comment]
Same issue

**Comment by wessberg** [comment]
Hey there. Since we already have tests for path mapping that are passing across environments, I'm interested in learning under which circumstances it is failing. You didn't provide much to go on, but I've tried to come up with the following example based on your code:

tsconfig.json:
```json
{
	"compilerOptions": {
		"outDir": "dist",
		"baseUrl": ".",
		"paths": {
			"@themes": [
				"themes"
			],
			"@themes/*": [
				"themes/*"
			]
		}
	}
}
```

index.ts:

```typescript
import {foo} from "@themes";
console.log(foo);
```

themes.ts:
```typescript
export const foo = 2;
```
rollup.config.js:
```javascript
import ts from "@wessberg/rollup-plugin-ts";

export default {
	input: "index.ts",
	output: {
		dir: "dist"
	},
	plugins: [
		ts({})
	]
}
```

On Mac OS 10.15.4, Node v13.5.0, that compiles just fine, with no errors.
Can you provide me with a minimal repro that will fail so I can investigate?

**Comment by antikvarBE** [comment]
Hi @wessberg,

Sorry for being cryptic, indeed. The reason I didn't provide much info is that my setup is actually not so different from what you've used above with the major difference of baseUrl being 'src'. Having said that, here's some more information that'll, hopefully, cast some light:

I'm currently building an external UI kit containing React components, fonts and SCSS stylesheets for use in CRA and Next.js apps. Relative imports within the UI kit itself work just fine and I'm sure I just can't see some silly mistake on my side.

---
Project tree (simplified)
```bash
.
├── CHANGELOG.md
├── LICENSE
├── README.md
├── package.json
├── rollup.config.js
├── src
│   ├── components
│   ├── fonts
│   ├── global.d.ts
│   ├── index.ts
│   └── styles
├── tsconfig.json
└── yarn.lock
```

---
tsconfig.json
```json
{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "allowJs": true,
    "jsx": "react",
    "declaration": true,
    "declarationDir": "dist",
    "sourceMap": true,
    "outDir": "dist",
    "removeComments": false,
    "noEmit": true,
    "isolatedModules": true,
    "strict": true,
    "noImplicitAny": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "moduleResolution": "node",
    "baseUrl": "src",
    //"paths": {"@components": ["components"], "@components/*": ["components/*"]} /* This is what I'd like to use. */,
    "rootDirs": [
      "src"
    ],
    "typeRoots": [
      "src",
      "node_modules/@types"
    ],
    "types": ["jest"],
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "skipLibCheck": true
  },
  "exclude": ["node_modules", "**/*.test.tsx", "**/*.stories.tsx"],
  "include": ["src"]
}
```

---
Rollup configuration

```typescript
import { DEFAULT_EXTENSIONS } from '@babel/core';
import babel from '@rollup/plugin-babel';
import commonjs from '@rollup/plugin-commonjs';
import resolve from '@rollup/plugin-node-resolve';
import replace from '@rollup/plugin-replace';
import typescript from '@wessberg/rollup-plugin-ts';
import copy from 'rollup-plugin-copy';
import postcss from 'rollup-plugin-postcss';
import { terser } from 'rollup-plugin-terser';
import pkg from './package.json';

const PLUGINS = [
  typescript(),
  postcss({
    autoModules: true,
  }),
  resolve(),
  babel({
    extensions: [...DEFAULT_EXTENSIONS, '.ts', '.tsx'],
    exclude: 'node_modules/**',
    babelHelpers: 'runtime',
  }),
  replace({
    'process.env.NODE_ENV': JSON.stringify('production'),
    _VERSION: JSON.stringify(pkg.version),
  }),
  copy({
    targets: [
      { src: 'src/styles/**/*', dest: 'dist/styles' },
      { src: 'src/fonts/**/*', dest: 'dist/fonts' },
    ],
  }),
  commonjs(),
];
export default [
  {
    input: 'src/index.ts',
    external: [...Object.keys({ ...pkg.dependencies, ...pkg.peerDependencies })],
    output: [
      {
        file: pkg.main,
        format: 'cjs',
      },
      {
        file: pkg.module,
        format: 'es',
      },
    ],
    plugins: [...PLUGINS],
  },
  {
    input: 'src/index.ts',
    external: [...Object.keys({ ...pkg.dependencies, ...pkg.peerDependencies })],
    output: [
      {
        name: 'ZvookUIKit',
        file: pkg['browser:min'],
        format: 'umd',
        globals: {
          classnames: 'callnames',
          react: 'React',
          'react-dom': 'ReactDOM',
          'react-infinite-scroll-hook': 'useInfiniteScroll',
        },
      },
    ],
    plugins: [...PLUGINS, terser()],
  },
];
```

---
Usage example

```typescript
import { Button } from '@components';
```
---
Error
```bash
(!) Unresolved dependencies
https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency
@components (imported by src/components/Jumbotron/Jumbotron.tsx)
[!] (plugin Typescript) TS2307: Cannot find module '@components' or its corresponding type declarations.
src/components/Jumbotron/Jumbotron.tsx (2:24)

2 import { Button } from '@components';
```

---
Also, for some reason, sourceMap files are not being generated despite the option being activated in tsconfig.json.

**Comment by wessberg** [comment]
Have you verified by that neither tsc nor the TypeScript language service reports the same diagnostic?

Regarding SourceMaps: You're in Rollup-land, so you'll  need to pass sourcemap: true as an output option to Rollup ☺️ .

**Comment by antikvarBE** [comment]
Lovely! I'm still relatively new to Rollup. I specified sourcemap: true and got my source maps the way I wanted them.

As of running tsc: no errors are being produced with the aforementioned example. VS Code Output for TypeScript doesn't yield any issues either.

**Comment by antikvarBE** [comment]
I forgot to share my .babelrc just in case:

```json
{
  "presets": ["@babel/preset-env", "@babel/react", "@babel/preset-typescript"],
  "plugins": [
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-transform-runtime"
  ]
}
```

**Comment by bastienrobert** [comment]
usually, i use [@rollup/plugin-alias](https://www.npmjs.com/package/@rollup/plugin-alias) 

```js
import alias from '@rollup/plugin-alias';
import tsconfig from './tsconfig.json';

function resolveEntries() {
  return Object.entries(
    tsconfig.compilerOptions.paths
  ).map(([find, [replacement]]) => ({ find, replacement }));
}

export default {
// ...
  plugins: [
    alias({
      resolve: ['.ts', '.tsx'],
      entries: resolveEntries(),
    }),
  ]
}
```

**Comment by marjorg** [comment]
@antikvarBE @wessberg Got the same issue here, did you guys resolve it?

Doesn't look like @rollup/plugin-alias works in this case.

**Comment by wessberg** [comment]
@bastienrobert, @marjorg, @antikvarBE, @sami616: Given that there are several passing tests related to path mapping in general in `rollup-plugin-ts`, including in combination with `@rollup/plugin-alias`, I will need a repro to be able to investigate this further. Thank you for the details, as well as the rollup and tsconfig examples, but since these are quite similar to the current test cases, I welcome one of you to prepare a repro that I can clone locally with a reproducable error such that I can prepare a fix against it.

**Comment by bastienrobert** [comment]
@wessberg I don't know if I really understood the problem, in any case I think it could be useful: https://github.com/bastienrobert/rollup-ts-paths-alias

**Comment by wessberg** [comment]
@bastienrobert, this looks like an example of how to combine the two plugins. That's great, but just to be avoid any confusion, I will need a link to a reproduction from someone who experiences issues with path mapping where I can reproducere one or more of the bugs that are described in this issue so that I can locate and address whatever underlying bug that may be present here :-)

**Comment by marjorg** [comment]
@wessberg I can give you access to our private repo if that's alright?

**Comment by wessberg** [comment]
Yeah, that could work! So long as there's a simple way to reproduce the issue, and you can provide me with instructions, that could work.

**Comment by marjorg** [comment]
@wessberg Invited you now. Checkout the only active feature branch and run `npm run build` in `/packages/components`

**Comment by wessberg** [comment]
Alright everyone, this issue has been fixed as part of v1.2.28 which will be released shortly. The problem was that `rollup-plugin-ts` always forced the `baseUrl` to `.`, which made path mapping resolve to wrong paths for other values for the `baseUrl`. @marjorg, I've tested it against the feature branch of your repo, and it built successfully. I don't need access to the repo anymore. Thanks, that made everything a lot easier!
